### PR TITLE
Opti 644 correct best cost

### PR DESCRIPTION
**Pull request by safraeli**

### CODE REVIEW

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java [hunk 1]
```diff
@@ -107,7 +107,11 @@ private boolean checkPercentageTerminationCondition(SearchStrategy.DiscoveredSol
             return false;
 
         boolean unassignedJobsEqual = (currentJobsUnassigned == unassignedJobsCount.get(i - noIterationWithoutImprovement));
-        boolean progressTooSlow = 100 * ((costs.get(i - noIterationWithoutImprovement) - currentCost) / currentCost)  <= terminationByCostPercentage;
-        return (unassignedJobsEqual && progressTooSlow);
+        boolean progressTooSlow = 100 * ((costs.get(i - noIterationWithoutImprovement) - bestCost) / bestCost)  <= terminationByCostPercentage;
+        if (unassignedJobsEqual && progressTooSlow){
+            log.debug("Termination condition by percentage reached after " + Integer.toString(i) + " Iterations: \n" + this);
+            return true;
+        }else
```
Comment (roir) [inline jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: no need for else just return outside the if block

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java [hunk 2]
```diff
@@ -107,7 +107,11 @@ private boolean checkPercentageTerminationCondition(SearchStrategy.DiscoveredSol
             return false;
 
         boolean unassignedJobsEqual = (currentJobsUnassigned == unassignedJobsCount.get(i - noIterationWithoutImprovement));
-        boolean progressTooSlow = 100 * ((costs.get(i - noIterationWithoutImprovement) - currentCost) / currentCost)  <= terminationByCostPercentage;
-        return (unassignedJobsEqual && progressTooSlow);
+        boolean progressTooSlow = 100 * ((costs.get(i - noIterationWithoutImprovement) - bestCost) / bestCost)  <= terminationByCostPercentage;
```
Comment (roir) [inline jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: feature changed but i don't see that the tests changed hich means this is not covered

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java (hunk 1 above)
Comment (safraeli) [reply jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: Done

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java (hunk 2 above)
Comment (safraeli) [reply jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: The feature remain the same, it was a logical implementation mistake.

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java [hunk 3]
```diff
@@ -107,7 +107,11 @@ private boolean checkPercentageTerminationCondition(SearchStrategy.DiscoveredSol
...
+        if (unassignedJobsEqual && progressTooSlow){
+            log.debug("Termination condition by percentage reached after " + Integer.toString(i) + " Iterations.");
```
Comment (hilaBringg) [inline jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: iterations (no need for capital i)

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java (hunk 2 above)
Comment (roir) [reply jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: sure, what i meant is that if we changed something in the logic, there should have been a change in the test as well, unless... there there is no test to cover this path of code/logic. 
please add

Comment (safraeli) [reply jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: Done

File: jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java (hunk 3 above)
Comment (safraeli) [reply jsprit-core/src/main/java/com/graphhopper/jsprit/core/algorithm/termination/IterationWithoutImprovementTermination.java]: Done

**Pull request merged**